### PR TITLE
WINDUP-593 - Unparsables report 

### PR DESCRIPTION
**Pull request by OndraZizka**
Test data are in #734

Finished the Unparsable Files report as it only had few things remaining to get it work.
It needs a cleanup and style fix. Which I will do later as I am switching to sprint tasks.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-core-PR-dependents-check/513/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-pr-builder/1141/

**Comment by OndraZizka** [comment]
Getting back to this soon

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/591/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1219/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/592/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1220/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1221/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/593/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1222/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/594/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1259/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/633/

**Comment by jsight** [comment]
What is the status of this PR?

**Comment by jsight** [comment]
There is a lot of commented out code in this and there are no tests. CreateUnparsableFilesReportRuleProvider2 has a lot of empty stub methods, and does not appear to produce a report with actual data.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1295/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/668/

**Comment by OndraZizka** [comment]
Rebased. This report now does not appear in the results. Need to check why.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/670/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1298/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1314/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/687/

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/UnparsablesAppReportModel.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @Adjacency(label = LABEL_PROJECT)
+    Iterable<ProjectModel> getProjects();
+
+    /**
+     * Files that had problems while parsing.
+     */
+    //        + ".out('" + ProjectModel.PROJECT_MODEL_TO_FILE + "').has('" + FileModel.PARSE_ERROR + "')")
+
+
+    /// REMOVE
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/UnparsablesAppReportModel.java]: Should these be removed?

**Comment by jsight** [comment]
This code needs tests in order to be reviewed and merged.

File: forks/frames/src/main/java/com/tinkerpop/frames/annotations/gremlin/GremlinGroovyAnnotationHandler.java [hunk 2]
```diff
@@ -81,7 +84,14 @@ public Object processVertex(final GremlinGroovy annotation, final Method method,
                     throw new IllegalStateException("The returned object can not be framed: " + result.getClass());
                 }
             } else {
-                return result;
+                /*
+                if (result instanceof Iterable) {
+                    // For the cases like Table - Gremlin returns an iterable with 1 item.
+                    Iterable resIt = (Iterable) result;
+                    return (ClassUtilities.returnsIterable(method)) ? resIt : resIt.iterator().hasNext() ? resIt.iterator().next() : null;
+                } else */{
```
Comment (mbriskar) [inline forks/frames/src/main/java/com/tinkerpop/frames/annotations/gremlin/GremlinGroovyAnnotationHandler.java]: commented out code

**Comment by OndraZizka** [comment]
Working on tests.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1382/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/754/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/755/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1383/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1384/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/756/

**Comment by jsight** [comment]
Merged https://github.com/windup/windup/commit/11236016368b1892b77e44635426f881dbd68de8

**Pull request closed without merging**